app/src/test/java/timur/gilfanov/messenger/domain/usecase/user/ObserveAndApplyLocaleUseCaseTest.kt

```diff
@@ -0,0 +1,160 @@
+    )
+    private val logger = NoOpLogger()
+
+    @Test
+    fun `applies locale when language is emitted`() = runTest {
+        val identityRepository = IdentityRepositoryStub(Success(testIdentity))
+        val settingsRepository = SettingsRepositoryStub(
+            settings = Success(Settings(UiLanguage.German)),
+        )
+        val localeRepository = LocaleRepositoryStub()
+        val observeUiLanguage = ObserveUiLanguageUseCase(
+            identityRepository,
+            settingsRepository,
+            logger,
+        )
+        val useCase = ObserveAndApplyLocaleUseCase(observeUiLanguage, localeRepository, logger)
+
+        useCase().test {
+            awaitItem()
+            awaitComplete()
+        }
+
+        assertEquals(listOf(UiLanguage.German), localeRepository.appliedLocales.toList())
+    }
+
+    @Test
+    fun `applies multiple locale changes`() = runTest {
+        val identityRepository = IdentityRepositoryStub(Success(testIdentity))
+        val settingsRepository = SettingsRepositoryStub(
+            settingsFlow = flow {
+                emit(Success(Settings(UiLanguage.English)))
+                emit(Success(Settings(UiLanguage.German)))
+                emit(Success(Settings(UiLanguage.English)))
+            },
+        )
+        val localeRepository = LocaleRepositoryStub()
+        val observeUiLanguage = ObserveUiLanguageUseCase(
+            identityRepository,
+            settingsRepository,
+            logger,
+        )
+        val useCase = ObserveAndApplyLocaleUseCase(observeUiLanguage, localeRepository, logger)
+
+        useCase().test {
+            awaitItem()
+            awaitItem()
+            awaitItem()
+            awaitComplete()
+        }
+
+        assertEquals(
+            listOf(UiLanguage.English, UiLanguage.German, UiLanguage.English),
+            localeRepository.appliedLocales.toList(),
+        )
+    }
+
+    @Test
+    fun `skips duplicate language emissions`() = runTest {
+        val identityRepository = IdentityRepositoryStub(Success(testIdentity))
+        val settingsRepository = SettingsRepositoryStub(
+            settingsFlow = flow {
+                emit(Success(Settings(UiLanguage.English)))
+                emit(Success(Settings(UiLanguage.English)))
+                emit(Success(Settings(UiLanguage.German)))
+            },
+        )
+        val localeRepository = LocaleRepositoryStub()
+        val observeUiLanguage = ObserveUiLanguageUseCase(
+            identityRepository,
+            settingsRepository,
+            logger,
+        )
+        val useCase = ObserveAndApplyLocaleUseCase(observeUiLanguage, localeRepository, logger)
+
+        useCase().test {
+            awaitItem()
+            awaitItem()
+            awaitComplete()
+        }
+
+        assertEquals(
+            listOf(UiLanguage.English, UiLanguage.German),
+            localeRepository.appliedLocales.toList(),
+        )
```

Comment:
Test must be changed, because behavior must be changed.

```suggestion
    @Test
    fun `does not skips duplicate language emissions`() = runTest {
        val identityRepository = IdentityRepositoryStub(Success(testIdentity))
        val settingsRepository = SettingsRepositoryStub(
            settingsFlow = flow {
                emit(Success(Settings(UiLanguage.English)))
                emit(Success(Settings(UiLanguage.English)))
                emit(Success(Settings(UiLanguage.German)))
            },
        )
        val localeRepository = LocaleRepositoryStub()
        val observeUiLanguage = ObserveUiLanguageUseCase(
            identityRepository,
            settingsRepository,
            logger,
        )
        val useCase = ObserveAndApplyLocaleUseCase(observeUiLanguage, localeRepository, logger)

        useCase().test {
            awaitItem()
            awaitItem()
            awaitItem()
            awaitComplete()
        }

        assertEquals(
            listOf(UiLanguage.English, UiLanguage.English, UiLanguage.German),
            localeRepository.appliedLocales.toList(),
        )
```